### ISSUE TITLE
`@stylistic/space-infix-ops`: Unexpected property "ignoreTypes". Expected properties: "int32Hint"

### ISSUE THREAD

**Issue by NishargShah** [eslint-stylistic/eslint-stylistic#768]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

I read the documentation and based on that I configured all the rules but I am getting this weird issue in `V3.1.0`, I can't install latest version as it is ESM

### Error

```bash
Error: Key "rules": Key "@stylistic/space-infix-ops":
	Value {"int32Hint":false,"ignoreTypes":false} should NOT have additional properties.
		Unexpected property "ignoreTypes". Expected properties: "int32Hint".
```

### Reproduction

Install V3.1.0 and it will throw this error

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by keijokapp** [comment]
The online documentation is for 4.2.0 only. `ignoreTypes` was added in 4.0.0.

I strongly suggest migrating ESLint configuration to ESM. It's not as difficult or scary as it's often made it out to be. It doesn't require migrating the rest of the project. It should be trivial in most cases - change `require`-s to `import`-s, `module.exports` to `export default` and rename `eslint.config.js` to `eslint.config.mjs` . If you get stuck, drop by [ESLint's Discord server](https://eslint.org/chat).

**Comment by NishargShah** [comment]
Sure, let me check
